packages/core/package.json

```diff
@@ -2,12 +2,12 @@
   "name": "@aerogearservices/core",
   "version": "1.0.0",
   "description": "Core configuration parser for AeroGear services",
-  "types": "src/index.ts",
+  "types": "dist/types/index.d.ts",
   "author": "Aerogear <aerogear@googlegroups.com>",
   "license": "Apache-2.0",
-  "main": "src/",
+  "main": "dist/index.js",
```

Comment:
It seems that we can support a wide array of tools by pointing to the bundle from a `browser` field in package.json, so:

```json
{
"main": "dist/index.js"
"browser": "dist/bundle.js"
types: "dist/types/index.d.ts"
}
```

see https://github.com/defunctzombie/package-browser-field-spec, https://webpack.js.org/configuration/resolve/#resolve-mainfields and https://www.typescriptlang.org/docs/handbook/declaration-files/publishing.html

Reply:
@paolobueno I like this 👍We could also support something like `require('@aerogear/mobile-core/bundle')` similar to how lodash supports `require('lodash/filter')`?

I published a [TS module](https://github.com/evanshortiss/binance.js) and just had a `prepublish` hook that executed `tsc` to create `d.ts` and `js` files beside the JS files, so is webpack necessary? If we need to generate a UMD bundle browserify works well [like this](https://github.com/evanshortiss/binance.js/blob/master/package.json#L13). I don't have much webpack experience so it might be doing something better that I'm not aware of 👍

Reply:
>  published a TS module and just had a prepublish hook that executed tsc to create d.ts and js files, so is webpack necessary

Yeah. This is how it works for us as well, but we also want to produce final bundle for users who want to use this for pure web platforms. 

I do not have strong opinion about solution, but over the last days I have been testing different options like gulp with browserify etc. Webpack in our case is also monitoring the source code and compiling them on the fly. We can use it for optimization etc. We went with something slightly bigger to avoid any in-house implementations in the future. It will be nicer for contributors to develop and live reload their examples etc.

Reply:
> Yeah. This is how it works for us as well, but we also want to produce final bundle for users who want to use this for pure web platforms.

Yep, that's what I was suggesting browserify for. Might not have been clear.

> Webpack in our case is also monitoring the source code and compiling them on the fly
`watchify` and `tsc -w` are also possible to watch code

Whatever is best for devexp and makes contributing easier gets my 👍 So if webpack makes all that easier, awesome.

Reply:
@evanshortiss do you recommend a UMD output yourself? I guess that target + typings file could be the easiest way to support running everywhere. Having only the minified bundle in the published package, so the end result would be awesome!

`require('@aerogear/mobile-core/bundle')` is different in intent than `require('lodash/filter')`.
The former is required for web targets if the main/browser field in package.json points to something the browser can't handle, like commonjs, that we want to avoid by having the browser field and/or maybe the UMD output.
The latter is an attempt at allowing optional optimization in which web code written in commonjs going through a compilation process could avoid adding the code for the whole `lodash` lib by only requiring a part of it. This can be done transparently in es6 modules by [tree shaking](https://webpack.js.org/guides/tree-shaking/). Anyway for the JS SDK each module will already be a separate one in the `@aerogear` npm scope

Reply:
> @evanshortiss do you recommend a UMD output yourself?

Sure. I think most developers expect this if they get a `bundle.js`.

This current output (in screenshot in PR) looks great. And appears to solve all scenarios really.